carta/image.py

```diff
@@ -593,6 +593,181 @@ def hide_contours(self):
+        """Set the vector overlay configuration, styling and colour or colourmap; and apply vector overlay; in a single step.
+
+        If both a colour and a colourmap are provided, the colourmap will be visible.
+
+        Parameters
+        ----------
+        angular_source : {0}
+            The angular source. If the image is with Stoke information, ``Computed PA`` is set by default; If the image is without Stoke information, ``Current image`` is set.
+        intensity_source : {1}
+            The intensity source. If the image is with Stoke information, ``Computed PI`` is set by default; If the image is without Stoke information, ``Current image`` is set.
+        pixel_averaging : {2}
+            The pixel averaging width in pixel. Set to ``None`` to disable pixel averaging.
+        fractional_intensity : {3}
+            Enable fractional polarization intensity. By default the absolute polarization intensity is used.
+        threshold : {4}
+            The threshold in Jy/pixels. By default is 4.
+        q_error : {5}
+            The Stokes Q error in Jy/beam. Set both this and ``u_error`` to enable debiasing. Debiasing is disabled by default.
+        u_error : {6}
+            The Stokes U error in Jy/beam. Set both this and ``q_error`` to enable debiasing. Debiasing is disabled by default.
+            thickness : {0}
+            The line thickness in pixels. By default is 1.
+        intensity_min : {1}
+            The minimum value of intensity in Jy/pixel.
+        intensity_min : {2}
+            The maximum value of intensity in Jy/pixel.
+        length_min : {3}
+            The minimum value of line length in pixels. By default is 0.
+        length_min : {4}
+            The maximum value of line length in pixels. By default is 20.
+        rotation_offset : {5}
+            The rotation offset in degrees. By default is 0.
+        color : {0}
+            The color.
+        colormap : {0}
+            The colormap.
+        bias : {1}
+            The colormap bias.
+        contrast : {2}
+            The colormap contrast.
+        """
+        self.configure_vector_overlay(angular_source, intensity_source, pixel_averaging, fractional_intensity, threshold, q_error, u_error)
+        self.set_vector_overlay_style(thickness, intensity_min, intensity_max, length_min, length_max, rotation_offset)
+        if color is not None:
+            self.call_action("vectorOverlayConfig.setColor", color)
+        if colormap is not None:
+            self.call_action("vectorOverlayConfig.setColormap", colormap)
+            self.call_action("vectorOverlayConfig.setColormapEnabled", True)
+            if bias is not None:
+                self.call_action("vectorOverlayConfig.setColormapBias", bias)
+            if contrast is not None:
+                self.call_action("vectorOverlayConfig.setColormapContrast", contrast)
+        else:
+            self.call_action("vectorOverlayConfig.setColormapEnabled", False)
```

Comment:
I realised that I made this unnecessarily complicated in my `plot_contours` PR, and have just updated it to reflect this. I was avoiding disabling the colormap in `set_color` -- but it doesn't actually matter, as long as you call `set_color` first and then `set_colormap` (because if they are both present, the colormap is shown). So you can do the same here, and first call `set_color` (if `color` is not `None`) and then `set_colormap` (if `colormap` is not `None`).

Reply:
Modification is made in 26aac1f.